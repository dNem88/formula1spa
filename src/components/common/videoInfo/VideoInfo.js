import React from 'react'
import styles from './videoInfo.module.css'
import {useNavigate} from 'react-router-dom'

function VideoInfo(props) {
    let navigate = useNavigate()
    function clickHandler() {
        navigate(`/videos/${props.id}`)
    }
    return (
        <div onClick={clickHandler} className={styles['light-info']}>
            <p className={styles['video-par']}>VIDEO</p>
            <p className={styles['content-video-par']}>{props.title}</p>
        </div>  
    )
}

export default VideoInfo;