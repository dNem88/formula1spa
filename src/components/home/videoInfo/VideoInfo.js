import React from 'react'
import styles from './videoInfo.module.css'
function VideoInfo(props) {
    return (
        <div className={styles['light-info']}>
            <p className={styles['video-par']}>VIDEO</p>
            <p className={styles['content-video-par']}>{props.title}</p>
        </div>  
    )
}

export default VideoInfo;