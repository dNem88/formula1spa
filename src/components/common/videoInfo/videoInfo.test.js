import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import VideoInfo
from './VideoInfo'

const mockProps = {
    title: 'test-title',
    id: 'test-id',
}

describe('test VideoInfo', () => {
    it('Snapshot test VideoInfo', () => {
        let component = renderer.create(
            <MemoryRouter>
                <VideoInfo id={mockProps.id} title={mockProps.title} />
            </MemoryRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
        expect(component.type).toBe('div')
        expect(component.props.className)
.toBe('light-info')   
        expect(component.children.length).toBe(2)
        expect(component.children[0].type).toBe('p')
        expect(component.children[0].props.className).toBe('video-par')
        expect(component.children[0].children[0]).toBe('VIDEO')
        expect(component.children[1].type).toBe('p')
        expect(component.children[1].props.className).toBe('content-video-par')
        expect(component.children[1].children[0]).toBe('test-title')
 })
})