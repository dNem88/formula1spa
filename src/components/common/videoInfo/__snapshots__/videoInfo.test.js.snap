// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test VideoInfo Snapshot test VideoInfo 1`] = `
<div
  className="light-info"
  onClick={[Function]}
>
  <p
    className="video-par"
  >
    VIDEO
  </p>
  <p
    className="content-video-par"
  >
    test-title
  </p>
</div>
`;
